packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -23,6 +23,17 @@ const restLength = template(`
   ARGUMENTS.length <= OFFSET ? 0 : ARGUMENTS.length - OFFSET
 `);
 
+function referencesRest(path, state) {
+  if (path.node.name === state.name) {
+    const ident = path.scope.getBindingIdentifier(state.name);
+    if (!ident || ident === state.outerBinding) {
+      return true;
+    }
```

Comment:
Thanks for your PR, it's looking good :+1:

Nit, `referencesRest` does not consistently return a `boolean`, do you mind adding a `return false` here?